[Ethan Chen Ee Shuen] iP

bipbipboopboop opened the pull request:
# Changee

> Click here to see the introduction of Changee

**Chagee** is **a simple to use task managing chat app**. You can tell **Chagee** to execute **commands** via **Command Line Interface** (CLI) or a **Graphical User Interface** (GUI).

Benefits of **Changee**:
- Create different types of tasks
- Show all tasks
- Delete tasks
- Mark as complete/incomplete
- Find tasks
- GUI

Roadmap:

- [X] Implement CLI
- [X] Implement GUI

You can also contribute to **Changee**. Locate `Laucher.java`, and start from this:

```java
public static void main(String[] args);
```

Commit:
Add Gradle support

bipbipboopboop committed:
Implement Duke Level-1 : Greet users, echo their input and exits upon typing bye

bipbipboopboop committed:
Implement Duke Level-2 : Implement add tasks and list all tasks

bipbipboopboop committed:
Implement Duke Level-3 : Implement mark as done

bipbipboopboop committed:
Implement basic ToDo, Event, Deadlines functionality

bipbipboopboop committed:
Implement Level-4 and Level-5 : Create ToDo, Event, Deadline and Error Handling

bipbipboopboop committed:
Fix tasks not printing after creation

bipbipboopboop committed:
Renames Tasks.java to TaskList.java and add Storage.java

Jnjy commented on src/main/java/tasks/Tasks.java:
```diff
@@ -0,0 +1,36 @@
+package tasks;
+
+public class Tasks {
```
Perhaps a more intuitive class name here?

Suggestion: `TaskList`

Jnjy commented on src/main/java/tasks/Tasks.java:
```diff
@@ -0,0 +1,36 @@
+
+public class Tasks {
+    private int numTasks = 0;
+    private Task[] tasksArray = new Task[100];
```
Perhaps you can use a more dynamic array here: `ArrayList<T>`

Jnjy commented on src/main/java/DukeParser.java:
```diff
@@ -0,0 +1,116 @@
+            }
+        }
+        // If command is invalid:
+        throw new Error("☹ OOPS!!! I'm sorry, but I don't know what that means :-(");
```
Any reason why you use the `Error` class instead of creating an `Exception`?

Jnjy commented on src/main/java/DukeCommand.java:
```diff
@@ -0,0 +1,10 @@
+public enum DukeCommand {
+    BYE("bye"), MARK("mark"), UNMARK("unmark"), LIST("list"), TODO("todo"), DEADLINE(
+            "deadline"), EVENT("event");
```
I like how you extracted out possible commands and make it into an `Enum` class!

Jnjy commented on src/main/java/DukeParser.java:
```diff
@@ -0,0 +1,116 @@
+            case LIST:
+            case BYE:
+                break;
+
```
Any reason there is a breakline here?

Jnjy reviewed (commented):
I like how you adhere to Coding Standard in the early stage of IP. More extractions can be done to improve the OOP of the code.

bipbipboopboop committed:
Implement Storage.java to store TaskList and Driver.java to process commands

winsonheng commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,112 @@
+        System.out.println(INDENTATION + "Hello! I'm Duke");
+        System.out.println(INDENTATION + "What can I do for you?");
+        System.out.println(BAR);
+        System.out.print(NEW_LINE);
```
Would it be more intuitive to use `System.out.println()` with no arguments instead?

winsonheng commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,112 @@
+    }
+
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
```
Should this be a final static field instead?

winsonheng commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,112 @@
+                    }
+                }
+                // System.out.println("Success! Your command is " + command.text);
+            } catch (Error e) {
```
Should you try to catch some other (and more specific) Exceptions instead? Maybe create a custom one? Errors generally indicate serious problems and it's probably [best not to attempt to catch them](https://docs.oracle.com/javase/7/docs/api/java/lang/Error.html).

winsonheng commented on src/main/java/DukeParser.java:
```diff
@@ -0,0 +1,116 @@
+    private String[] parseCommandArgs() {
+        List<String> commandArgs = new ArrayList<>();
+        switch (command) {
+            case LIST:
```
Maybe consider adding a TODO comment to indicate that the code here is not ready?

bipbipboopboop committed:
Add date string validation

bipbipboopboop committed:
Merge branch-Level-7 with branch-Level-8

bipbipboopboop committed:
Fix Task not Serializable bug and create more organized packages

bipbipboopboop committed:
Implement delete functionality

bipbipboopboop committed:
Reorganize packages and extract classes

bipbipboopboop committed:
Implement find task

bipbipboopboop committed:
Add javadocs to commmonly used public methods

bipbipboopboop committed:
Fix merge conflict between branch-Level-9 and branch-A-Javadoc

bipbipboopboop committed:
Merge remote-tracking branch 'upstream/add-gradle-support'

bipbipboopboop committed:
Add gradle checkstyle

bipbipboopboop committed:
Refactor Ui and Driver print messages

bipbipboopboop committed:
Refactor Driver, fix event parsing bug and test text-ui

bipbipboopboop committed:
Fix build.gradle file

bipbipboopboop committed:
Add some more javadocs and fix some identifier

bipbipboopboop committed:
Configure .vscode to check for coding standard

bipbipboopboop committed:
Merge branch 'branch-A-CodingStandard'

bipbipboopboop committed:
Set up JUnit

bipbipboopboop committed:
Refactor DukeParser

bipbipboopboop committed:
Add JUnit tests

bipbipboopboop committed:
Update TODO

bipbipboopboop committed:
Add basic GUI that echos user input

bipbipboopboop committed:
Implement GuiDriver to accomodate Gui

bipbipboopboop committed:
Refactor GuiDriver

bipbipboopboop committed:
Remove redundant files and refactor Ui.java into UiPrinter.java

bipbipboopboop committed:
DukeParser class have many similar methods within a large switch case.

These methods are too long to be fitted within a screen and they throw
different errors.

These methods can be extracted out and reorganized to increase readibility and
reusability.

We groups commands into 3 main categories
1) Task Creation commands : todo deadline event
2) Index-based commands : mark unmark delete
3) Misc : list, bye, find

Within these ecah categories, we put them together in the switch
statement, and make them throw the same error if the user has input an
invalid command.

bipbipboopboop committed:
Implement view schedule

bipbipboopboop committed:
Add assertion to check if commandArg is 0 for list and bye command

bipbipboopboop committed:
Merge pull request #2 from bipbipboopboop/branch-A-Assertions

Add assertion to check if commandArg is 0 for list and bye command

bipbipboopboop committed:
Fix checkstyle

bipbipboopboop committed:
Fix trailing whitespace checkstyle error and add exclude checkstyle to ignore switch default case

bipbipboopboop committed:
Merge pull request #3 from bipbipboopboop/branch-A-CodingStandard

Fix checkstyle and adhere to coding standard

bipbipboopboop committed:
Implement more error handling

bipbipboopboop committed:
Kill Duke and replace with Chagee

bipbipboopboop committed:
Fix checkstyle and code quality

bipbipboopboop committed:
Merge pull request #4 from bipbipboopboop/branch-A-CodeQuality

Fix checkstyle and code quality

bipbipboopboop committed:
Fix resources path not loading

bipbipboopboop committed:
Increase Chagee UI height and add Ui.png

bipbipboopboop committed:
Update User Guide

bipbipboopboop committed:
Rename wrong naming : Chagee.java to ChageeCommand.java
